src/integrated/soil_canopy_model.jl

```diff
@@ -310,18 +320,38 @@ function net_radiation_at_ground(
     c = FT(LSMP.light_speed(earth_param_set))
     h = FT(LSMP.planck_constant(earth_param_set))
     N_a = FT(LSMP.avogadro_constant(earth_param_set))
-    (; λ_γ_PAR, λ_γ_NIR) = canopy_radiation.parameters
+    (; λ_γ_PAR, λ_γ_NIR, ϵ_canopy) = canopy_radiation.parameters
     energy_per_photon_PAR = h * c / λ_γ_PAR
     energy_per_photon_NIR = h * c / λ_γ_NIR
-    SW_d_beneath_canopy = @. SW_d - (
-        (energy_per_photon_PAR * N_a * p.canopy.radiative_transfer.apar) +
-        (energy_per_photon_NIR * N_a * p.canopy.radiative_transfer.anir)
-    )
-    LW_d_beneath_canopy = LW_d # Assumes T_canopy = T_air
-    return @. (
-        -(1 - p.α_soil) * SW_d_beneath_canopy -
-        p.ϵ_soil * (LW_d_beneath_canopy - _σ * p.T_soil^4)
-    )
+    T_canopy =
+        ClimaLSM.Canopy.canopy_temperature(canopy.energy, canopy, Y, p, t)
+    T_soil = p.T_soil
+    α_soil = p.α_soil
+    ϵ_soil = p.ϵ_soil
+    APAR = p.canopy.radiative_transfer.apar
+    ANIR = p.canopy.radiative_transfer.anir
+
+    LW_net_canopy = p.canopy_LW_n
+    SW_net_canopy = p.canopy_SW_n
+    LW_net_soil = p.soil_LW_n
+    SW_net_soil = p.soil_SW_n
+    LW_out = p.LW_out
+    SW_out = p.SW_out
+
+
+    LW_d_canopy = @. (1 - ϵ_canopy) * LW_d + ϵ_canopy * _σ * T_canopy^4 # double checked
+    LW_u_soil = @. ϵ_soil * _σ * T_soil^4 + (1 - ϵ_soil) * LW_d_canopy # double checked
+    @. LW_net_canopy =
+        ϵ_canopy * LW_d - 2 * ϵ_canopy * _σ * T_canopy^4 + ϵ_canopy * LW_u_soil # double checked
+    @. LW_net_soil = ϵ_soil * LW_d_canopy - ϵ_soil * _σ * T_soil^4 # double checked
+    @. LW_out = (1 - ϵ_canopy) * LW_u_soil * ϵ_canopy * _σ * T_canopy^4 # double checked
+
+    @. SW_net_canopy =
```

Comment:
Sw_net_canopy = absorbed - reflected - transmitted 

these 3 terms should come from the two-stream scheme. 

SW_d_canopy = transmitted

Reply:
Thank you!